src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8682,13 +8682,18 @@ enum Precedence : uint
             Cast,
             PointerIndirection,
             AddressOf,
-            Primary_UNUSED, // Primaries are parsed in an ad-hoc manner.
+            Primary,
         }
 
         private static Precedence GetPrecedence(SyntaxKind op)
         {
             switch (op)
             {
+                case SyntaxKind.QueryExpression:
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.AnonymousMethodExpression:
+                    return Precedence.Expression;
```

Comment:
I feel like I'm missing something, are these not meant to be Lambda precedence?

Reply:
Yes.  However, since `Expression` is not used anywhere else, it won't have an observable or testable effect on the behavior.

---
In reply to: [357866628](https://github.com/dotnet/roslyn/pull/40089#discussion_r357866628) [](ancestors = 357866628)

Reply:
Moreover, they are technically at the same precedence.  See [https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#expression](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#expression)

---
In reply to: [357875828](https://github.com/dotnet/roslyn/pull/40089#discussion_r357875828) [](ancestors = 357875828,357866628)

Reply:
Makes sense, thank you!

Reply:
I'm going to merge the three precedence levels in the next iteration unless that runs into problems!

---
In reply to: [357879626](https://github.com/dotnet/roslyn/pull/40089#discussion_r357879626) [](ancestors = 357879626)